.github/copilot-instructions.md

```diff
@@ -0,0 +1,147 @@
+# curl-http3-deb
+
+curl-http3-deb creates .deb packages for curl with HTTP/3 and QUIC support for Ubuntu 22.04/24.04. It packages the build instructions from curl.se/docs/http3.html into installable .deb packages and distributes them via Docker Hub and a custom apt repository.
+
+Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## Working Effectively
+
+### Docker Build Process
+- Multi-stage Docker build with stages: `prereqs` -> `build` -> `curl` -> `deploy`
+- **NEVER CANCEL builds** - compilation takes 45+ minutes. ALWAYS set timeouts to 60+ minutes minimum.
+- Build process:
+  1. `prereqs` stage: Install build tools (takes ~37 seconds)
+  2. `build` stage: Compile OpenSSL, nghttp3, ngtcp2, curl from source (takes 45+ minutes total)
+  3. `curl` stage: Create runtime Docker image with compiled curl
+  4. `deploy` stage: Publish .deb packages to Gemfury apt repository
+
+### Build Commands and Timing
+- `docker build --target prereqs -t curl-prereqs .` -- takes 37 seconds
+- `docker build --target build -t curl-build .` -- takes 45+ minutes. NEVER CANCEL. Set timeout to 60+ minutes.
+- `docker build --target curl -t curl-final .` -- takes 45+ minutes. NEVER CANCEL. Set timeout to 60+ minutes.
+- `docker build .` -- full build takes 45+ minutes. NEVER CANCEL. Set timeout to 60+ minutes.
+
+### Network Dependencies and Known Issues
+- **CRITICAL**: Build process requires network access to clone from GitHub:
+  - `https://github.com/quictls/openssl` (OpenSSL with QUIC support)
+  - `https://github.com/ngtcp2/nghttp3` (HTTP/3 library)
+  - `https://github.com/ngtcp2/ngtcp2` (QUIC library)
+  - `https://github.com/curl/curl` (curl source)
+- **Known Issue**: Docker builds may fail with SSL certificate verification errors in restricted network environments
+- **Workaround**: Use pre-built Docker image `compscidr/curl-http3-quic:latest` for testing and validation
```

Comment:
This is not a good workaround - the whole point of the smoke test, and this entire repo is to build the debian package of curl with http3 included in it. If the build fails - that is a problem which must be fixed.